File: Baekjoon/15000~15999/[15720]카우버거/solution.py
from audioop import reverse


B, C, D = map(int, input().split())
max_size = max([B, C, D])
res1 = res2 = 0

burger = sorted(list(map(int, input().split())), reverse=True)
side = sorted(list(map(int, input().split())), reverse=True)
drink = sorted(list(map(int, input().split())), reverse=True)

total = [[] for i in range(max_size)]

for i in range(max_size):
    total[i] = [0, True]
    if i < B:
        total[i][0] += burger[i]
    else:
        total[i][1] = False
    if i < C:
        total[i][0] += side[i]
    else:
        total[i][1] = False
    if i < D:
        total[i][0] += drink[i]
    else:
        total[i][1] = False

for i in range(max_size):
    res1 += total[i][0]

    res2 += int(total[i][0] * 0.9) if total[i][1] == True else total[i][0]

print(res1)
print(res2)

File: Baekjoon/15000~15999/[15720]카우버거/solution.cpp
#include <algorithm>
#include <iostream>
#include <vector>

#define endl "\n"

using namespace std;

typedef pair<int, bool> pib;

vector<int> burger;
vector<int> side;
vector<int> drink;
vector<pib> total;
int B, C, D, max_size, res1, res2;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  cin >> B >> C >> D;

  max_size = max({B, C, D});

  burger = vector<int>(B);
  side = vector<int>(C);
  drink = vector<int>(D);
  total = vector<pib>(max_size, {0, true});
  res1 = res2 = 0;

  for (int i = 0; i < B; i++) {
    cin >> burger[i];
  }

  for (int i = 0; i < C; i++) {
    cin >> side[i];
  }

  for (int i = 0; i < D; i++) {
    cin >> drink[i];
  }

  sort(burger.begin(), burger.end(), greater<>());
  sort(side.begin(), side.end(), greater<>());
  sort(drink.begin(), drink.end(), greater<>());

  for (int i = 0; i < max_size; i++) {
    if (i < B) {
      total[i].first += burger[i];
    } else {
      total[i].second = false;
    }
    if (i < C) {
      total[i].first += side[i];
    } else {
      total[i].second = false;
    }
    if (i < D) {
      total[i].first += drink[i];
    } else {
      total[i].second = false;
    }
  }

  for (int i = 0; i < max_size; i++) {
    res1 += total[i].first;

    if (total[i].second) {
      res2 += total[i].first * 0.9;
    } else {
      res2 += total[i].first;
    }
  }

  cout << res1 << endl;

  cout << res2 << endl;

  return 0;
}